components/email_aliases/email_aliases_service.cc

```diff
@@ -318,4 +385,155 @@ std::string EmailAliasesService::GetAuthTokenForTesting() const {
+      std::move(user_callback).Run(base::ok(ok->alias));
+      return;
+    }
+  }
+  std::move(user_callback)
+      .Run(base::unexpected(l10n_util::GetStringUTF8(
+          IDS_EMAIL_ALIASES_ERROR_ALIAS_NOT_AVAILABLE)));
+}
+
+void EmailAliasesService::OnUpdateAliasResponse(
+    UpdateAliasCallback user_callback,
+    std::optional<std::string> response_body) {
+  RefreshAliases();
+  auto parsed = ParseResponseDictOrError(response_body);
+  if (!parsed.has_value()) {
+    std::move(user_callback).Run(base::unexpected(parsed.error()));
+    return;
+  }
+  if (auto edited = AliasEditedResponse::FromValue(parsed.value())) {
+    if (edited->message != "updated") {
+      std::move(user_callback).Run(base::unexpected(edited->message));
+      return;
+    }
+  }
+  std::move(user_callback).Run(base::ok(std::monostate{}));
+}
+
+void EmailAliasesService::OnDeleteAliasResponse(
+    DeleteAliasCallback user_callback,
+    std::optional<std::string> response_body) {
+  RefreshAliases();
+  auto parsed = ParseResponseDictOrError(response_body);
+  if (!parsed.has_value()) {
+    std::move(user_callback).Run(base::unexpected(parsed.error()));
+    return;
+  }
+  if (auto edited = AliasEditedResponse::FromValue(parsed.value())) {
+    if (edited->message != "deleted") {
+      std::move(user_callback).Run(base::unexpected(edited->message));
+      return;
+    }
+  }
+  std::move(user_callback).Run(base::ok(std::monostate{}));
+}
+
+void EmailAliasesService::RefreshAliases() {
+  ApiFetch(email_aliases_service_base_url_.Resolve("?status=active"),
+           net::HttpRequestHeaders::kGetMethod,
+           base::BindOnce(&EmailAliasesService::OnRefreshAliasesResponse,
+                          weak_factory_.GetWeakPtr()));
+}
+
+void EmailAliasesService::OnRefreshAliasesResponse(
+    std::optional<std::string> response_body) {
+  // TODO(https://github.com/brave/brave-browser/issues/48959):
+  // In this function, when an error happens, we should show
+  // an error message to the user (requires design work)
+  if (!response_body) {
+    LOG(ERROR) << "Email Aliases service error: No response body";
+    return;
+  }
+  auto parsed = base::JSONReader::Read(*response_body);
+  if (parsed && parsed->is_dict() && parsed->GetDict().contains("message")) {
+    auto* error_message = parsed->GetDict().FindString("message");
+    if (error_message) {
+      LOG(ERROR) << "Email Aliases service error: " << *error_message;
+      return;
+    }
+  }
+  if (!parsed || !parsed->is_list()) {
+    LOG(ERROR) << "Email Aliases service error: Invalid response format";
+    return;
+  }
+  std::vector<email_aliases::mojom::AliasPtr> aliases;
+  for (const auto& alias_val : parsed->GetList()) {
+    if (!alias_val.is_dict()) {
+      LOG(ERROR) << "Email Aliases service error: Invalid response format";
+      return;
+    }
+    if (auto entry = AliasListEntry::FromValue(alias_val.GetDict())) {
+      auto alias_obj = email_aliases::mojom::Alias::New();
```

Comment:
A bit out of scope, but this mojom type looks like:

```
struct Alias {
  // The alias email address for this alias
  string email;
  // Note attached by user
  string? note;
  // Domains where email is used
  array<string>? domains;
};
```

We don't seem to be using any of these fields, or setting them.